scipy/signal/setup.py

```diff
@@ -22,8 +23,17 @@ def configuration(parent_package='', top_path=None):
 
     config.add_extension(
         '_spectral', sources=['_spectral.c'])
-    config.add_extension(
-        '_max_len_seq_inner', sources=['_max_len_seq_inner.c'])
+
+    if int(os.environ.get('SCIPY_USE_PYTHRAN', 0)):
+        import pythran
+        ext = pythran.dist.PythranExtension(
+            'scipy.signal._max_len_seq_inner',
+            sources=["scipy/signal/_max_len_seq_inner.py"])
+        config.ext_modules.append(ext)
```

Comment:
Ah, I see now why the module name needs to include `scipy/signal/` - you're not calling `config.add_extension`, so things like subpackage handling are skipped.